### PR TITLE
Fix SPIRV friendly IR for OpImageSampleExplicitLod&OpSampledImage

### PR DESCRIPTION
**Pull request by Fznamznon**
These instructions were translated directly to OCL even if SPIR-V
friendly LLVM IR was requested.

### CODE REVIEW

**Comment by Fznamznon** [comment]
@AlexeySotkin , @AlexeySachkov , @MrSidims please take a look.

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -1280,60 +1282,6 @@ static char getTypeSuffix(Type *T) {
   return Suffix;
 }
 
-// ToDo: Handle unsigned integer return type. May need spec change.
```
Comment (AlexeySachkov) [inline lib/SPIRV/SPIRVReader.cpp:1283]: This `TODO` was lost, I suggest that we preserve it if it is still actual

Comment (Fznamznon) [reply lib/SPIRV/SPIRVReader.cpp:1283]: Ok, fair enough.

**Comment by Fznamznon** [comment]
Ping.

**Review by MrSidims — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Pull request merged**